SafeGraph_py_functions/SafeGraph_py_functions.py

```diff
@@ -113,65 +113,99 @@ def unpack_json_separate(df_, json_column='visitor_home_cbgs', key_col_name='vis
+      day_visits_exp[array_sequence] = day_visits_exp[array_sequence] -1
+    day_visits_exp.drop(['dummy_key'], axis=1, inplace=True)
+    day_visits_exp.rename(columns={array_column+'_json': new_col}, inplace=True)
+    day_visits_exp[new_col] = day_visits_exp[new_col].astype('int64')
+    df.drop([array_column+'_json'], axis=1, inplace=True)
+    df = pd.merge(df, day_visits_exp, on=[place_key,file_key])
+    return df
 
 ### ------------------------------------------ END JSON SECTION--------------------------------------------------------
 
-### ---------------------------------------CORE AND PATTERNS SECTION -----------------------------------------------
+### ---------------------------------------CORE, GEO, AND PATTERNS SECTION -----------------------------------------------
 
 
-def read_core_folder(path_to_core, use_cols=None):
+def read_core_folder(path_to_core, *args, **kwargs):
     core_files = glob.glob(os.path.join(path_to_core, "*.csv.gz"))
     print(f"You are about to load in {len(core_files)} core files")
 
     li = []
     for core in core_files:
         print(core)
-        df = pd.read_csv(core, usecols=use_cols, compression='gzip',
+        df = pd.read_csv(core, *args, **kwargs, compression='gzip',
                          dtype={'postal_code': str, 'phone_number': str, 'naics_code': str})
         li.append(df)
 
     SG_core = pd.concat(li, axis=0)
     return SG_core
 
+### added a new core read that takes the information straight from the zipped file (like you get it from the catelog)
 
-def read_pattern_demo(f_path, use_cols=None, compression='gzip', nrows=100):
-    df = pd.read_csv(f_path, dtype={'postal_code': str, 'phone_number': str, 'naics_code': str}, nrows=nrows,
-                     compression=compression,
-                     usecols=use_cols)
-    return df
+def read_core_folder_zip(path_to_core, *args, **kwargs):
+    zip_file = ZipFile(path_to_core)
+
+    li = []
+
+    dfs = {text_file.filename: pd.read_csv(zip_file.open(text_file.filename), *args, **kwargs, compression='gzip',
+                         dtype={'postal_code': str, 'phone_number': str, 'naics_code': str})
+           for text_file in zip_file.infolist()
+           if text_file.filename.endswith('.csv.gz')}
+
+    SG_core = pd.concat(dfs, axis=0, ignore_index=True)
+
+    return SG_core
+
+def read_geo_zip(path_to_zip, *args, **kwargs):
+  zf = zp.ZipFile(path_to_zip)
+  result=pd.read_csv(zf.open('core_poi-geometry.csv.gz'), compression='gzip', *args, **kwargs)
+
+  return result
 
 
-def read_pattern_single(f_path, use_cols=None, compression='gzip'):
-    df = pd.read_csv(f_path, dtype={'postal_code': str, 'phone_number': str, 'naics_code': str}, usecols=use_cols,
-                     compression=compression)
+def read_pattern_single(f_path, *args, **kwargs):
+    df = pd.read_csv(f_path, dtype={'postal_code': str, 'phone_number': str, 'naics_code': str}, *args, **kwargs,
+                     compression='gzip')
```

Comment:
@Trippl7777  is there a reason you went back to hard-coding compression = 'gzip' instead of letting it be one of the free parmeters captured by kwargs, or its own param with default = 'gzip'?

Reply:
The hard coding compression is because I want it to be the default. We get a lot of questions from people about how to open gz files. Do you think it should just be standard and expect the user to add their compression parameter? Does safegraph have an non .gz files?

Reply:
I think it should be how it was before. `compression` is a configurable parameter that defaults to the correct answer for the intended use. but if for some reason the files aren't gziped, the user can override the default using the `compression` parameter.

Reply:
in either case, we expect 99% of the time users to not include the `compression` parameter in their function call. so the only difference is that we support the 1% of cases where a user might want to override the default.

Reply:
where default means `gzip` for our function